src/Microsoft.AspNetCore.Authentication.OAuth/OAuthHandler.cs

```diff
@@ -206,16 +207,15 @@ protected virtual string BuildChallengeUrl(AuthenticationProperties properties,
             var scope = FormatScope();
 
             var state = Options.StateDataFormat.Protect(properties);
-
-            var queryBuilder = new QueryBuilder()
+            var parameters = new Dictionary<string, string>
             {
                 { "client_id", Options.ClientId },
                 { "scope", scope },
                 { "response_type", "code" },
                 { "redirect_uri", redirectUri },
                 { "state", state },
             };
-            return Options.AuthorizationEndpoint + queryBuilder.ToString();
+            return QueryHelpers.AddQueryString(Options.AuthorizationEndpoint, parameters);
```

Comment:
We've done this in other places in the auth middleware, we could also do it here. https://github.com/aspnet/Security/blob/5231c5a853a36f62bcd4f982660a59b2b7f1dd40/src/Microsoft.AspNetCore.Authentication.Facebook/FacebookHandler.cs#L28